chatbot-core/data/preprocessing/preprocess_plugin_docs.py

```diff
@@ -0,0 +1,37 @@
+import json
+import os
+from bs4 import BeautifulSoup
+from utils import (
+    remove_tags,
+    remove_html_comments,
+    get_visible_text_length
+)
+
```

Comment:
- Consider parsing and re-serializing with BeautifulSoup to avoid potential malformed HTML affecting filtering.
- Add more informative and consistent logging like `print(f"[INFO] Processed {len(processed_plugin_docs)} out of {len(plugin_data)} plugin docs.")`(just an example)